### PR TITLE
dynamic OPC fees

### LINKED ISSUE #476: Dynamic Ocean Community Consume fee
Currently, it's hardcoded as 1%.
Let's have it defined in FactoryRouter, just like any other fees

### PR DESCRIPTION
**Pull request by alexcos20**
Closes #476

Changes proposed in this PR:

- Factory:  allow OPF to set consumeFee & providerFee
- Default values:     consumeFee - 1%  , providerFee - 0%
- ERC20Template:  query & use that cut for both

PS: need tests, but blocked right now because tests are not available

### CODE REVIEW

**Comment by trentmc** [comment]
> PS: need tests, but blocked right now because tests are not available

What's stopping writing tests?

**Comment by alexcos20** [comment]
@lacoop6tu needs to rewrite all tests, seems that hardhat has a new issue now and everything related to forking is not working any more

File: artifacts/contracts/interfaces/IFactoryRouter.sol/IFactoryRouter.json [hunk 1]
```diff
@@ -104,6 +104,19 @@
       "stateMutability": "nonpayable",
       "type": "function"
     },
+    {
+      "inputs": [],
+      "name": "getConsumeFees",
```
Comment (trentmc) [inline artifacts/contracts/interfaces/IFactoryRouter.sol/IFactoryRouter.json:109]: Shouldn't it be "fee" not "fees" everywhere?

File: contracts/interfaces/IFactoryRouter.sol [hunk 2]
```diff
@@ -18,6 +18,8 @@ interface IFactoryRouter {
     ) external returns (bytes32 exchangeId);
 
     function getOPFFee(address baseToken) external view returns (uint256);
```
Comment (trentmc) [inline contracts/interfaces/IFactoryRouter.sol:20]: Shouldn't it be "OPC" not "OPF" everywhere? (Is there a different github issue for that?)

**Review by trentmc — CHANGES_REQUESTED** [review]
At a glance, everything looks sane. I had a couple minor comments.

I will feel much more comfortable when we have tests for this.

**Comment by trentmc** [comment]
> @lacoop6tu needs to rewrite all tests, seems that hardhat has a new issue now and everything related to forking is not working any more

Is there a github issue for this?

**Comment by alexcos20** [comment]
> > @lacoop6tu needs to rewrite all tests, seems that hardhat has a new issue now and everything related to forking is not working any more
> 
> Is there a github issue for this?

https://github.com/oceanprotocol/contracts/issues/478

File: contracts/interfaces/IFactoryRouter.sol (hunk 2 above)
Comment (alexcos20) [reply contracts/interfaces/IFactoryRouter.sol:20]: I will update it in this PR

File: artifacts/contracts/interfaces/IFactoryRouter.sol/IFactoryRouter.json (hunk 1 above)
Comment (alexcos20) [reply artifacts/contracts/interfaces/IFactoryRouter.sol/IFactoryRouter.json:109]: fixed

**Review by lacoop6tu — APPROVED** [review]

**Pull request merged**